server/routes/createGoal/createGoalController.ts

```diff
@@ -24,7 +25,27 @@ export default class CreateGoalController {
       } as NewGoal
     }
 
-    const view = new CreateGoalView(prisonerSummary, req.session.newGoal.createGoalForm, req.flash('errors'))
+    const futureGoalTargetDates = [
+      {
```

Comment:
I really like the idea of `futureGoalTargetDates` being an array of `{ text: string; value: string }` and t being a field on the view model object. Nice 👍 

I wonder if we could factor some of this out into a separate module function, and then that can be unit tested separately.
I was thinking something like this here in the controller:
```
const futureGoalTargetDates = [
    someFunctionCall(moment().toDate(), 3),
    someFunctionCall(moment().toDate(), 6),
    someFunctionCall(moment().toDate(), 12),
]
```
I've deliberately called it `someFunctionCall` because I'm not sure what we want to call it yet (I'll let you have a think about that 😁 )

And then the implementation of `someFunctionCall` might be something like:
```
export default someFunctionCall = (startDate: Date, additionalMonths: number): { text: string, value: string } => {
    const futureDate = moment(startDate).add(additionalMonths, 'months').format('D-MMMM-YYYY`)
    return {
        text: futureDate,
        value: `in ${additionalMonths} months (${futureDate})`,
    }
}
```
Something like that .... however ....

It also needs to cater for the stupid rule where if today is the 29/30/31st, and the future month does not have 29/30/31 days, then it needs to be "rolled back" to the last day of the month.
This is where your unit tests will come into play 👍 

Have a think and see what you think

Reply:
Thanks @nathanrussell-moj-digital will take a look at that.

As for the final point around days in a month, `moment` is smart enough to handle this itself..

> Special considerations for months and years
> If the day of the month on the original date is greater than the number of days in the final month, the day of the month will change to the last day in the final month.

https://momentjscom.readthedocs.io/en/latest/moment/03-manipulating/01-add/

But we can definitely add tests on our end to make sure this is true

Reply:
Oh, fantastic news! `moment` is so smart 😁 Saves us writing what would otherwise be some pretty f'ugly code! 👍 

Re: unit tests, defo include one that would trigger a 29th Feb leap year. 2024 is the next leap year, so perhaps a test where the start date is 30/11/2023, add 3 months, expect output to be 29/02/2024

Something like that anyway 👍

Reply:
Added the shared function and tests. The tests do work because a) I spelt February wrong in the tests despite being born in Feburary b) Miscalculated 6 months from February - the tests caught both!

Also the test `should return the last day of month, if the day of the month on the original date is greater than the number of days in the final month` covers both the scenarios we discussed above; the leap year but also the if the future month does not have the same amount of days it rolls back. Hopefully 1 test for both is good enough, rather than duplicating it.